pkg/cloud/ibmcloud/actuators/machine/actuator.go

```diff
@@ -176,6 +176,13 @@ func (ic *IbmCloudClient) Exists(ctx context.Context, cluster *clusterv1.Cluster
 		return false, err
 	}
 
+	if (guest != nil) && (machine.Spec.ProviderID == nil || *machine.Spec.ProviderID == "") {
+		// TODO(xunpan): this does not work in ibm cloud
```

Comment:
`this does not work in ibm cloud`
you mean this function not work or not work in Exists function ?maybe open a futher issue to track this ?

Reply:
This code is found in other providers. But it does not work in ibmcloud: with this code, the `ProviderID` is not updated as expected.

My code in `Create` works. It updates `ProviderID`. I keep code here to try to understand if there has any misunderstanding in updating `ProviderID` in `Exists`. This is not a issue but need further understanding of the code. The full code logic should works for any case because I check the `ProviderID` if it is specified already.

Reply:
thank you for the detailed info, really appreciate, that works for me